### PR TITLE
Handle 'no combination' specific price submit

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When creating a specific price in Product BO page, if "apply to all combinations" was selected it would not work. I fixed it by using default value zero.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/18751
| How to test?  | See ticket

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19845)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]
I aproved the PR and won't block it for such a nitpick, but this would be easier to read imho: 

```php
$id_product_attribute = $specificPriceValues['sp_id_product_attribute'] ?? 0;
```
Not sure if we have a guideline for this kind of thing? Maybe it's just my personal preference lately.

**Comment by matks** [comment]
> I aproved the PR and won't block it for such a nitpick, but this would be easier to read imho:
> 
> ```
> $id_product_attribute = $specificPriceValues['sp_id_product_attribute'] ?? 0;
> ```
> 
> Not sure if we have a guideline for this kind of thing? Maybe it's just my personal preference lately.

That is a relevant feedback 😊 actually

File: src/Adapter/Product/AdminProductWrapper.php [hunk 1]
```diff
@@ -268,7 +268,7 @@ public function processDependsOnStock(Product $product, $dependsOnStock, $forAtt
     public function processProductSpecificPrice($id_product, $specificPriceValues, $idSpecificPrice = null)
     {
         // ---- data formatting ----
-        $id_product_attribute = $specificPriceValues['sp_id_product_attribute'];
+        $id_product_attribute = isset($specificPriceValues['sp_id_product_attribute']) ? $specificPriceValues['sp_id_product_attribute'] : 0;
```
Comment (matks) [inline src/Adapter/Product/AdminProductWrapper.php]:
```suggestion
        $id_product_attribute = $specificPriceValues['sp_id_product_attribute'] ?? 0;
```

**Comment by matks** [comment]
@matthieu-rolland I updated the PR 😉

**Review by matthieu-rolland — DISMISSED** [review]

File: src/Adapter/Product/AdminProductWrapper.php [hunk 2]
```diff
@@ -268,7 +268,7 @@ public function processDependsOnStock(Product $product, $dependsOnStock, $forAtt
     public function processProductSpecificPrice($id_product, $specificPriceValues, $idSpecificPrice = null)
     {
         // ---- data formatting ----
-        $id_product_attribute = $specificPriceValues['sp_id_product_attribute'];
+        $id_product_attribute = $specificPriceValues['sp_id_product_attribute'] ?? 0;
         $id_shop = $specificPriceValues['sp_id_shop'] ? $specificPriceValues['sp_id_shop'] : 0;
         $id_currency = $specificPriceValues['sp_id_currency'] ? $specificPriceValues['sp_id_currency'] : 0;
         $id_country = $specificPriceValues['sp_id_country'] ? $specificPriceValues['sp_id_country'] : 0;
```
Comment (PierreRambaud) [inline src/Adapter/Product/AdminProductWrapper.php]:
```suggestion
        $id_shop = $specificPriceValues['sp_id_shop'] ?? 0;
        $id_currency = $specificPriceValues['sp_id_currency'] ?? 0;
        $id_country = $specificPriceValues['sp_id_country'] ?? 0;
```

Not blocking the PR but, should be more consistent :thinking: use the same way everywhere :sweat_smile: 

With :heart:

Comment (matks) [reply src/Adapter/Product/AdminProductWrapper.php]: Let's do this together 💪

**Review by PululuK — DISMISSED** [review]

File: src/Adapter/Product/AdminProductWrapper.php (hunk 2 above)
Comment (PululuK) [reply src/Adapter/Product/AdminProductWrapper.php]: Good evening @matks 
Could you do the same on the line [275](https://github.com/matks/PrestaShop/blob/fix-specific-price-bad-index/src/Adapter/Product/AdminProductWrapper.php#L275) please. 😅
Thank you man

Comment (matks) [reply src/Adapter/Product/AdminProductWrapper.php]: And when I do it, someone else will pop up and ask me to do it again on another line ? 😛 is that your plan ?

Comment (PululuK) [reply src/Adapter/Product/AdminProductWrapper.php]: "Make prestashop consistent again ..." 😅

Comment (matks) [reply src/Adapter/Product/AdminProductWrapper.php]: (Luigi's voice) "Here we go !"
https://github.com/PrestaShop/PrestaShop/pull/19845/commits/12f779e467f7a066a40a2f2155a1609068581c99

**Comment by SD1982** [comment]
Hi @matks I still have an error but it's not the same this time 
capture d'écran_81

**Comment by matks** [comment]
@SD1982 The code suggestions of my friends have broken my code 😛 
I fix this issue in another PR where I remove the suggestions => https://github.com/PrestaShop/PrestaShop/pull/19865

I knew I should stick to my scope 😅 my fault

**Comment by SD1982** [comment]
Ok bo no problem I will assign myself to the new PR !

**Pull request closed without merging**